core/tls-websocket-server/src/tls_common.rs

```diff
@@ -41,42 +35,26 @@ pub fn make_config(cert: &str, key: &str) -> WebSocketResult<Arc<rustls::ServerC
 	Ok(Arc::new(config))
 }
 
-fn load_certs(filename: &str) -> WebSocketResult<Vec<rustls::Certificate>> {
-	let certfile = fs::File::open(filename).map_err(|e| {
-		WebSocketError::InvalidCertificate(format!(
-			"Failed to load certificate from file ({}): {:?}",
-			filename, e
-		))
-	})?;
-	let mut reader = BufReader::new(certfile);
-	rustls::internal::pemfile::certs(&mut reader).map_err(|_| {
-		WebSocketError::InvalidCertificate("Failed to parse certificate file".to_string())
-	})
+fn load_certs(content: &str) -> WebSocketResult<Vec<rustls::Certificate>> {
+	let mut reader = BufReader::new(content.as_bytes());
+	rustls::internal::pemfile::certs(&mut reader)
+		.map_err(|_| WebSocketError::InvalidCertificate("Failed to parse certificate".to_string()))
```

Comment:
One solution for the `DER` -> `PEM` -> `DER` conversion could be, to use (in `initialization.rs`):
```rust
let cert_der: Vec<u8> = cert.serialize_der().map_err(..)?;
```
to get the encoded `DER` certificate. And then use that directly to construct a `rustls::Certificate` (no need for a vector here I think, since it's only a single certificate?):
```rust
let certificate: rustls::Certificate = rustls::Certificate(cert_der); 
```
I haven't tested this myself, it's just an idea - maybe you can test it out? It would also change the types we pass into the web-server config from `String`/`str` to an encoded vector, `Vec<u8>`.

Reply:
It is worth testing this to change from an rcgen::Certificate to a rustls::Certificate. But don't forget that we should support  certificate chain and not just a single root certificate

Reply:
I had to do a _bit_ of research, but in a nutshell the intermediate `PEM` representation is needed in this case*.

-----

*: 
Basically `PEM` is a list of `base64` encoded `DER` objects (like a container). [RFC](https://www.rfc-editor.org/rfc/rfc7468#section-5.1)
And `DER` is the encoding used for the binary representation of `ASN.1` defined structures.

As long the `PEM` format contains only one object it can be converted to `DER` (and vice-versa).
But when there are multiple certificates involved (in case of a certificate chain) then only `PEM` can be used.
(Java [supports](https://docs.oracle.com/javase/7/docs/api/java/security/cert/CertificateFactory.html#generateCertificates(java.io.InputStream)) the _multiple certificates in a single `DER` object_ case, but I think it might be their own addition as `OpenSSL` [does not seem to have this feature](https://docs.rs/openssl/latest/openssl/x509/struct.X509.html#implementations) and I could not find it in any standard either.)

Reply:
Thanks for your research and explanations.

Reply:
Alright, thanks for finding that out ☺️ So we'll leave it as is, that's totally fine for me as well. Maybe you can add a comment somewhere in the code to summarize what we discussed here

Reply:
I would keep the intermediate PEM representation. 
For the moment the tls code gives very little information about the required and possible certificates (PEM, DER, Chain, File, ....). I don't think we should improve this in this PR, but at least add some comments in this file.

Reply:
Thank you, added a comment regarding it in https://github.com/integritee-network/worker/pull/1034/commits/8c68a23792aeee9e9a0985ad5ed898b4ed07f89a